### PR TITLE
Deploy notebook fix

### LINKED ISSUE #88: deploy notebook feature broken due to api change
the deploy notebook button is broken because the API has changed

the new endpoint is: https://cloud-prod.ploomber.io/

you can look at `/docs` to get the documentation. the first deployment should work fine by just updating the host, but I think re-deployment will break. there use to be one endpoint for first time deployment and another one for re-deployment but there's only one now

### PR DESCRIPTION
**Pull request by neelasha23**
## Describe your changes

1. Modified ploomber cloud endpoint
2. Added `project_id` to endpoint when redeploying.

## Issue number

Closes #88 

## Checklist before requesting a review

- [ ] Performed a self-review of my code
- [ ] Formatted my code with [`pkgmt format`](https://ploomber-contributing.readthedocs.io/en/latest/contributing/pr.html#linting-formatting)
- [ ] Added [tests](https://ploomber-contributing.readthedocs.io/en/latest/contributing/pr.html#testing) (when necessary).
- [ ] Added [docstring](https://ploomber-contributing.readthedocs.io/en/latest/contributing/pr.html#documenting-changes-and-new-features) documentation and update the [changelog](https://ploomber-contributing.readthedocs.io/en/latest/contributing/pr.html#changelog) (when needed)

### CODE REVIEW

**Comment by neelasha23** [comment]
This is the output when `Deploy Notebook` is clicked second time after first deployment. Subsequent clicks on the button deploys the notebook again. Please confirm if this is the expected behaviour @edublancas 

Screenshot 2023-11-28 at 5 55 37 PM

**Comment by edublancas** [comment]
@neelasha23 did you verify that the notebook is successfully deployed on ploomber cloud? and that modifying the notebook and clicking again successfully redeploys?

File: src/dialog.tsx [hunk 1]
```diff
@@ -156,16 +158,42 @@ export const DialogContent = (props: any): JSX.Element => {
...
+        }).then(reply => {
+            var result = reply["project_details"]
+            var errorMsg: {
+                type: string,
+                detail: any
+            } = {
+                type: "generic",
+                detail: ""
+            }
+            console.log(result)
```
Comment (edublancas) [inline src/dialog.tsx:184]: remove this console.log

**Review by edublancas — CHANGES_REQUESTED** [review]

File: src/dialog.tsx (hunk 1 above)
Comment (neelasha23) [reply src/dialog.tsx]: I removed  the entire code block as it was unnecessary

**Comment by neelasha23** [comment]
1. In the first deployment project is getting created , I've verified the dashboard successfully deployed.
2. Even if there are no changes in the notebook and `Deploy Notebook` is clicked it creates a fresh deployment with a new project ID. I don't think there is a check to detect if notebook got changed?

@edublancas

**Comment by edublancas** [comment]
> Even if there are no changes in the notebook and Deploy Notebook is clicked it creates a fresh deployment with a new project ID. I don't think there is a check to detect if notebook got changed?

we haven't implemented logic to detect if the notebook is changed so that's expected. what's not expected is that if you deploy the notebook and then click again, it should keep the same project ID. check the logic, I think it might be due to changes in the API

> Also, there are a lot of 404 errors in the CI and the tests are timing out and eventually failing. The deploy notebook tests have passed in my local.

yeah we have some intermittent failures but looks like the CI passed now

**Comment by neelasha23** [comment]
Fixed the code. Now it's redeploying to the same `project_id`. PFB screenshots for a sample notebook:

Screenshot 2023-11-29 at 5 12 02 PM

Screenshot 2023-11-29 at 5 27 05 PM

@edublancas

File: src/dialog.tsx (hunk 1 above)
Comment (edublancas) [reply src/dialog.tsx]: don't forget to include a link to the changes

Comment (neelasha23) [reply src/dialog.tsx]: Found the commit: https://github.com/ploomber/jupysql-plugin/pull/90/commits/ccc584b2968e20daa1164de1c14ae055653800a6c

**Comment by neelasha23** [comment]
Where should I add changelog entry? `0.2.6` or create a new header? @edublancas

**Review by edublancas — APPROVED** [review]

**Comment by edublancas** [comment]
you need to create a new header section. but no worries, I'll take it from here

**Pull request merged**